apps/system/js/nfc_handover_manager.js

```diff
@@ -266,6 +201,59 @@
       this.bluetoothStatusSaved = false;
     },
 
+    /**
+     * Get adapter from bluetooth through promise interface.
+     * Since BTv1 get onenabled event before onadapteradded event,
+     * We need wait a certain time to get adapter after
+     * onenabled event is retrieved.
+     *
+     * @return {Promise}
+     * @memberof NfcHandoverManager.prototype
+     */
+    _retrieveAdapter: function _retrieveAdapter() {
+      return new Promise((resolve, reject) => {
+        var adapter = Service.query('Bluetooth.getAdapter');
+        if (adapter !== null) {
+          this._debug('First hand BT adapter retrieved');
+          resolve(adapter);
+        } else {
+          setTimeout(() => {
```

Comment:
Yes, please use Service.request() in client side and Service.register in server side.